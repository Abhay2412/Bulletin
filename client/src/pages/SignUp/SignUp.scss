@import "../../styles/colors";

.signup-container {
  min-height: 100vh;
  height: 100vh;
  .signup {
    height: 100%;
    width: 100%;

    &-app-description {
      color: $color-white;
      text-align: center;
      margin-top: -3px;
    }

    &-left-container {
      background-color: $color-geyser;
      height: 100%;
      padding: 50px 0px;
    }

    &-app-name {
      color: $color-lynch;
      margin-top: -5px;
    }

    &-signup-redirect-text {
      color: $color-lynch;
    }

    &-signup-redirect-button {
      background-color: $color-pink;
      color: $color-white;
      margin-top: -10px;
      width: 40%;
      min-width: 150px;
    }

    &-text-field {
      margin-top: -15px;
      width: 100%;
      min-width: 200px;
    }

    &-title {
      color: $color-brown;
    }

    &-button {
      background-color: $color-pink;
      color: $color-brown;
      width: 40%;
      min-width: 150px;
      margin-top: 50px;
    }

    &-inner-grid {
        width: 100%;
    }

    &-inner-grid-item {
        width: 60%;
        min-width: 200px;
    }
  }
}
