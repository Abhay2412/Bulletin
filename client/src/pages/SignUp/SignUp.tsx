import axios from "axios";
import axiosInstance from "../../axios";
import React, { useState } from "react";
import { Grid, Button, TextField } from "@material-ui/core";
import Logo from "../../assets/logo.svg";
import { signUpEndPoint } from "../../authEndPoints";
import "./SignUp.scss";

interface SignUpErrors {
  name: string;
  email: string;
  password: string;
  confirmPassword: string;
}

const SignUp: React.FC = () => {
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfrimPassword] = useState("");
  const [errors, setErrors] = useState<SignUpErrors>({
    name: "",
    email: "",
    password: "",
    confirmPassword: "",
  });

  const validateData = () => {
    errors.name = "";
    errors.email = "";
    errors.password = "";
    errors.confirmPassword = "";

    let errorsExits = false;

    if (!name) {
      errors.name = "Please enter your name.";
      errorsExits = true;
    }

    if (!email) {
      errors.email = "Please enter your email.";
      errorsExits = true;
    } else if (
      !/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/.test(
        email
      )
    ) {
      errors.email = "Please enter a valid email.";
      errorsExits = true;
    }

    if (!password) {
      errors.password = "Please enter your password.";
      errorsExits = true;
    }

    if (!confirmPassword) {
      errors.confirmPassword = "Please re-enter your password.";
      errorsExits = true;
    } else if (password != confirmPassword) {
      errors.confirmPassword = "Passwords do not match.";
      errorsExits = true;
    }

    setErrors({ ...errors });
    return !errorsExits;
  };

  const handleSignup = () => {
    if (!validateData()) {
      return;
    }

    axios
      .post(
        signUpEndPoint,
        { email: email, password: password, returnSecureToken: true },
        {
          headers: {
            "content-type": "application/json",
          },
        }
      )
      .then((res) => {
        console.log("Account Created!!!");
        console.log("Sign Up response is: ", res);
        axiosInstance
          .post("/addUser", {
            id: res.data.localId,
            name: name,
            email: email,
            birthDay: null,
            alternativeEmail: "",
            phoneNumber: "",
            overview: "",
            boards: [],
          })
          .then((res) => {
            console.log("user added to data base.");
            console.log("response is: ", res);
          })
          .catch((err) => {
            console.log("failed to add user to database", err);
          });
      })
      .catch((err) => {
        errors.email = "Email already exists.";
        setErrors({ ...errors });

        console.log("Failed to sign up user: ", err);
      });
  };

  const handleEmailChange = (
    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    setEmail(event.target.value);
  };

  const handlePasswordChange = (
    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    setPassword(event.target.value);
  };

  const handleNameChange = (
    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    setName(event.target.value);
  };

  const handleConfirmPasswordChange = (
    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    setConfrimPassword(event.target.value);
  };

  return (
    <div className="signup-container">
      <Grid className="signup" direction="row" container spacing={2}>
        <Grid
          className="signup-left-container"
          container
          item
          md={5}
          lg={5}
          xl={5}
        >
          <Grid
            className="signup-inner-grid"
            justifyContent="center"
            alignItems="center"
            container
            item
            xs={12}
          >
            <Grid
              className="signup-inner-grid"
              container
              item
              direction="column"
              justifyContent="center"
              alignItems="center"
              xs={12}
            >
              <img className="signup-logo-image" src={Logo} />
              <h1 className="signup-app-name">Bulletin</h1>
              <h3 className="signup-app-description">
                Organize your household in a smart way{" "}
              </h3>
            </Grid>
            <Grid
              className="signup-inner-grid"
              container
              item
              direction="column"
              justifyContent="center"
              alignItems="center"
              xs={12}
            >
              <h4 className="signup-signup-redirect-text">
                Already have an account?
              </h4>
              <Button
                className="signup-signup-redirect-button"
                variant="contained"
              >
                Log in
              </Button>
            </Grid>
          </Grid>
        </Grid>
        <Grid
          className="signup-inner-grid"
          container
          justifyContent="center"
          alignItems="center"
          item
          xs={12}
          md={7}
          lg={7}
          xl={7}
          direction="column"
          spacing={2}
        >
          <Grid item>
            <h1 className="signup-title">Sign Up</h1>
          </Grid>
          <Grid className="signup-inner-grid-item" item>
            <p>Name</p>
            <TextField
              className="signup-text-field"
              InputLabelProps={{ shrink: false }}
              value={name}
              onChange={handleNameChange}
              label={name === "" ? "Enter your full name" : ""}
              variant="outlined"
              error={errors.name != ""}
              helperText={errors.name}
            />
          </Grid>
          <Grid className="signup-inner-grid-item" item>
            <p>Email</p>
            <TextField
              className="signup-text-field"
              InputLabelProps={{ shrink: false }}
              value={email}
              onChange={handleEmailChange}
              label={email === "" ? "Enter your email address" : ""}
              variant="outlined"
              error={errors.email != ""}
              helperText={errors.email}
            />
          </Grid>
          <Grid className="signup-inner-grid-item" item>
            <p>Password</p>
            <TextField
              className="signup-text-field"
              InputLabelProps={{ shrink: false }}
              value={password}
              onChange={handlePasswordChange}
              label={password === "" ? "Enter your password" : ""}
              variant="outlined"
              type="password"
              error={errors.password != ""}
              helperText={errors.password}
            />
          </Grid>
          <Grid className="signup-inner-grid-item" item>
            <p>Confrim Password</p>
            <TextField
              className="signup-text-field"
              InputLabelProps={{ shrink: false }}
              value={confirmPassword}
              onChange={handleConfirmPasswordChange}
              label={confirmPassword === "" ? "Re-enter your password" : ""}
              variant="outlined"
              type="password"
              error={errors.confirmPassword != ""}
              helperText={errors.confirmPassword}
            />
          </Grid>
          <Grid item>
            <Button
              onClick={handleSignup}
              className="signup-button"
              variant="contained"
            >
              Sign Up
            </Button>
          </Grid>
        </Grid>
      </Grid>
    </div>
  );
};

export default SignUp;
