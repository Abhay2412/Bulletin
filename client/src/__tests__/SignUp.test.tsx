import React from "react";
import SignUp from "../pages/SignUp/SignUp";
import { configure, mount, ReactWrapper } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("Testing <SignUp /> Component", () => {
  it("should change name field value", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );
    wrapper
      .find('input[type="text"]')
      .find("#signup-name-field")
      .simulate("change", {
        target: { value: "Test User" },
      });
    expect(
      wrapper.find('input[type="text"]').find("#signup-name-field").get(0).props
        .value
    ).toEqual("Test User");
  });

  it("should change email field value", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );
    wrapper
      .find('input[type="text"]')
      .find("#signup-email-field")
      .simulate("change", {
        target: { value: "test@domain.com" },
      });
    expect(
      wrapper.find('input[type="text"]').find("#signup-email-field").get(0)
        .props.value
    ).toEqual("test@domain.com");
  });

  it("should change password field value", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );
    wrapper
      .find('input[type="password"]')
      .find("#signup-password-field")
      .simulate("change", {
        target: { value: "12345678" },
      });
    expect(
      wrapper
        .find('input[type="password"]')
        .find("#signup-password-field")
        .get(0).props.value
    ).toEqual("12345678");
  });

  it("should change confrim password field value", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );
    wrapper
      .find('input[type="password"]')
      .find("#signup-confirm-password-field")
      .simulate("change", {
        target: { value: "12345678" },
      });
    expect(
      wrapper
        .find('input[type="password"]')
        .find("#signup-confirm-password-field")
        .get(0).props.value
    ).toEqual("12345678");
  });

  it("should have name error", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );

    wrapper.find("button").find(".signup-button").simulate("click");
    expect(
      wrapper.find("p").find("#signup-name-field-helper-text").text()
    ).toEqual("Please enter your name.");
  });

  it("should have email error", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );

    wrapper.find("button").find(".signup-button").simulate("click");
    expect(
      wrapper.find("p").find("#signup-email-field-helper-text").text()
    ).toEqual("Please enter your email.");
  });

  it("should have password error", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );

    wrapper.find("button").find(".signup-button").simulate("click");
    expect(
      wrapper.find("p").find("#signup-password-field-helper-text").text()
    ).toEqual("Please enter your password.");
  });

  it("should have confirm password error", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );

    wrapper.find("button").find(".signup-button").simulate("click");
    expect(
      wrapper
        .find("p")
        .find("#signup-confirm-password-field-helper-text")
        .text()
    ).toEqual("Please re-enter your password.");
  });

  it("should have confirm password error", () => {
    const wrapper: ReactWrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );

    wrapper
      .find('input[type="password"]')
      .find("#signup-password-field")
      .simulate("change", {
        target: { value: "12345678" },
      });

    wrapper
      .find('input[type="password"]')
      .find("#signup-confirm-password-field")
      .simulate("change", {
        target: { value: "145678" },
      });

    wrapper.find("button").find(".signup-button").simulate("click");
    expect(
      wrapper
        .find("p")
        .find("#signup-confirm-password-field-helper-text")
        .text()
    ).toEqual("Passwords do not match.");
  });
});
