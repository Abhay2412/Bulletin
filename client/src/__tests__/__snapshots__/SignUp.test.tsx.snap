// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SignUp /> Component should render properly 1`] = `
<div
  className="signup-container"
>
  <div
    className="MuiGrid-root signup MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root signup-left-container MuiGrid-container MuiGrid-item MuiGrid-grid-md-5 MuiGrid-grid-lg-5 MuiGrid-grid-xl-5"
    >
      <div
        className="MuiGrid-root signup-inner-grid MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center MuiGrid-grid-xs-12"
      >
        <div
          className="MuiGrid-root signup-inner-grid MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center MuiGrid-grid-xs-12"
        >
          <img
            className="signup-logo-image"
            src="logo.svg"
          />
          <h1
            className="signup-app-name"
          >
            Bulletin
          </h1>
          <h3
            className="signup-app-description"
          >
            Organize your household in a smart way
             
          </h3>
        </div>
        <div
          className="MuiGrid-root signup-inner-grid MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center MuiGrid-grid-xs-12"
        >
          <h4
            className="signup-signup-redirect-text"
          >
            Already have an account?
          </h4>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained signup-signup-redirect-button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Log in
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root signup-inner-grid MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-md-7 MuiGrid-grid-lg-7 MuiGrid-grid-xl-7"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h1
          className="signup-title"
        >
          Sign Up
        </h1>
      </div>
      <div
        className="MuiGrid-root signup-inner-grid-item MuiGrid-item"
      >
        <p>
          Name
        </p>
        <div
          className="MuiFormControl-root MuiTextField-root signup-text-field"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="signup-name-field"
            id="signup-name-field-label"
          >
            Enter your full name
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="signup-name-field"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Enter your full name
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiGrid-root signup-inner-grid-item MuiGrid-item"
      >
        <p>
          Email
        </p>
        <div
          className="MuiFormControl-root MuiTextField-root signup-text-field"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="signup-email-field"
            id="signup-email-field-label"
          >
            Enter your email address
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="signup-email-field"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Enter your email address
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiGrid-root signup-inner-grid-item MuiGrid-item"
      >
        <p>
          Password
        </p>
        <div
          className="MuiFormControl-root MuiTextField-root signup-text-field"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="signup-password-field"
            id="signup-password-field-label"
          >
            Enter your password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="signup-password-field"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Enter your password
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiGrid-root signup-inner-grid-item MuiGrid-item"
      >
        <p>
          Confrim Password
        </p>
        <div
          className="MuiFormControl-root MuiTextField-root signup-text-field"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="signup-confirm-password-field"
            id="signup-confirm-password-field-label"
          >
            Re-enter your password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="signup-confirm-password-field"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Re-enter your password
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root signup-button css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Sign Up
        </button>
      </div>
    </div>
  </div>
</div>
`;
